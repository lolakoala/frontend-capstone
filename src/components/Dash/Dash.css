.toolbar-link {
  text-decoration: none;
  font-size: 20px;
  color: #315659;
  display: inline-block;
  margin-top: 15px;
}

.toolbar-link:hover {
  font-weight: bold;
  color: #2F9C95;
}

.dash-instruction {
  width: 100%;
  height: auto;
  text-align: center;
  background-color: #d4c9a9;
  padding-bottom: 30px;
  color: #315659;
}
