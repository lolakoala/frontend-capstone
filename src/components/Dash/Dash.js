import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Toolbar, Tabs, Tab } from 'material-ui';
import { Link } from 'react-router-dom';
import SearchProfessionals from '../SearchProfessionals/SearchProfessionals';
import SearchBuddies from '../SearchBuddies/SearchBuddies';

class Dash extends Component {
  componentWillMount() {
    this.props.clearSearchResults();
  }

  render() {
    const {
      getInsuranceList,
      getSpecialtyList,
      insuranceList,
      specialtyList,
      search,
      allChallenges,
      getAllChallenges,
      currentUser
    } = this.props;
    return (
      <div>
        <Toolbar>
          <Link to="/chat">Chat</Link>
          <Link to="/listUserProfs">My Preferred Proffesionals</Link>
          <Link to="/listUserBuddies">My Buddies</Link>
          {/* In My Toolbox, should we add ability to add your own resources? */}
          <Link to="/toolbox">My Toolbox</Link>
        </Toolbar>
        <Tabs>
          <Tab label="Search Professionals">
            <SearchProfessionals
              getInsuranceList={getInsuranceList}
              getSpecialtyList={getSpecialtyList}
              insuranceList={insuranceList}
              specialtyList={specialtyList}
              search={search}
              currentUser={currentUser}
            />
          </Tab>
          <Tab label="Search Buddies">
            <SearchBuddies
              allChallenges={allChallenges}
              getAllChallenges={getAllChallenges}
              search={search}
              currentUser={currentUser}
            />
          </Tab>
        </Tabs>
      </div>
    );
  }
}

export default Dash;

Dash.propTypes = {
  getInsuranceList: PropTypes.func,
  getSpecialtyList: PropTypes.func,
  insuranceList: PropTypes.array,
  specialtyList: PropTypes.array,
  search: PropTypes.func,
  clearSearchResults: PropTypes.func,
  allChallenges: PropTypes.array,
  getAllChallenges: PropTypes.func,
  currentUser: PropTypes.object
};
