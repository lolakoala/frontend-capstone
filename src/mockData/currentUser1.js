const currentUser1 = {
  email: 'lola@gmail.com',
  city: 'Denver',
  userName: 'lolakoala',
  aboutMe: 'Lola loves doggos.'
};

export default currentUser1;
