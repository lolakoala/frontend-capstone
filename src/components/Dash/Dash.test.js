import React from "react";
import Dash from './Dash.js';
import { shallow, configure } from "enzyme";
import currentUser1 from '../../mockData/currentUser1.js';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Dash', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<Dash
    getInsuranceList={mockFn}
    getSpecialtyList={mockFn}
    insuranceList={['medicaid', 'bcbs', 'aetna']}
    specialtyList={['couples', 'youth', 'gender']}
    search={mockFn}
    clearSearchResults={mockFn}
    allChallenges={['anxiety', 'depression']}
    getAllChallenges={mockFn}
    currentUser={currentUser1}
  />);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match snap', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a toolbar', () => {
    const toolBar = wrapper.find('Toolbar');

    expect(toolBar.length).toEqual(1);
  });

  it('should have 5 links', () => {
    const links = wrapper.find('Link');

    expect(links.length).toEqual(5);
  });

  it('should have tabs', () => {
    const tabs = wrapper.find('Tabs');

    expect(tabs.length).toEqual(1);
  });

  it('should have 2 inner tabs', () => {
    const tabs = wrapper.find('Tab');

    expect(tabs.length).toEqual(2);
  });

  it('should call function on mount', () => {
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
