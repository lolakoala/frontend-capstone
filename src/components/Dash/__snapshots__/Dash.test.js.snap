// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dash should match snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dash
    allChallenges={
        Array [
            "anxiety",
            "depression",
          ]
    }
    clearSearchResults={[Function]}
    currentUser={
        Object {
            "aboutMe": "Lola loves doggos.",
            "city": "Denver",
            "email": "lola@gmail.com",
            "userName": "lolakoala",
          }
    }
    getAllChallenges={[Function]}
    getInsuranceList={[Function]}
    getSpecialtyList={[Function]}
    insuranceList={
        Array [
            "medicaid",
            "bcbs",
            "aetna",
          ]
    }
    search={[Function]}
    specialtyList={
        Array [
            "couples",
            "youth",
            "gender",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Toolbar
          noGutter={false}
          style={
                    Object {
                              "backgroundColor": "#d4c9a9",
                            }
          }
>
          <Link
                    className="toolbar-link"
                    replace={false}
                    to="/chat"
          >
                    Chat
          </Link>
          <Link
                    className="toolbar-link"
                    replace={false}
                    to="/list/userProfs"
          >
                    My Preferred Professionals
          </Link>
          <Link
                    className="toolbar-link"
                    replace={false}
                    to="/list/userBuddies"
          >
                    My Buddies
          </Link>
          <Link
                    className="toolbar-link"
                    replace={false}
                    to="/toolbox"
          >
                    My Toolbox
          </Link>
          <Link
                    className="toolbar-link"
                    replace={false}
                    to="/profile"
          >
                    My Profile
          </Link>
</Toolbar>,
        <div
          className="dash-instruction"
>
          Use the tabs below to connect with professionals or buddies.
</div>,
        <Tabs
          initialSelectedIndex={0}
          onChange={[Function]}
>
          <Tab
                    label="Search Professionals"
          >
                    <SearchProfessionals
                              currentUser={
                                        Object {
                                                  "aboutMe": "Lola loves doggos.",
                                                  "city": "Denver",
                                                  "email": "lola@gmail.com",
                                                  "userName": "lolakoala",
                                                }
                              }
                              getInsuranceList={[Function]}
                              getSpecialtyList={[Function]}
                              insuranceList={
                                        Array [
                                                  "medicaid",
                                                  "bcbs",
                                                  "aetna",
                                                ]
                              }
                              profSearch={undefined}
                              pushHistory={[Function]}
                              search={[Function]}
                              specialtyList={
                                        Array [
                                                  "couples",
                                                  "youth",
                                                  "gender",
                                                ]
                              }
                    />
          </Tab>
          <Tab
                    label="Search Buddies"
          >
                    <SearchBuddies
                              allChallenges={
                                        Array [
                                                  "anxiety",
                                                  "depression",
                                                ]
                              }
                              buddySearch={undefined}
                              currentUser={
                                        Object {
                                                  "aboutMe": "Lola loves doggos.",
                                                  "city": "Denver",
                                                  "email": "lola@gmail.com",
                                                  "userName": "lolakoala",
                                                }
                              }
                              getAllChallenges={[Function]}
                              pushHistory={[Function]}
                              search={[Function]}
                    />
          </Tab>
</Tabs>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              className="toolbar-link"
              replace={false}
              to="/chat"
>
              Chat
</Link>,
            <Link
              className="toolbar-link"
              replace={false}
              to="/list/userProfs"
>
              My Preferred Professionals
</Link>,
            <Link
              className="toolbar-link"
              replace={false}
              to="/list/userBuddies"
>
              My Buddies
</Link>,
            <Link
              className="toolbar-link"
              replace={false}
              to="/toolbox"
>
              My Toolbox
</Link>,
            <Link
              className="toolbar-link"
              replace={false}
              to="/profile"
>
              My Profile
</Link>,
          ],
          "noGutter": false,
          "style": Object {
            "backgroundColor": "#d4c9a9",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Chat",
              "className": "toolbar-link",
              "replace": false,
              "to": "/chat",
            },
            "ref": null,
            "rendered": "Chat",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "My Preferred Professionals",
              "className": "toolbar-link",
              "replace": false,
              "to": "/list/userProfs",
            },
            "ref": null,
            "rendered": "My Preferred Professionals",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "My Buddies",
              "className": "toolbar-link",
              "replace": false,
              "to": "/list/userBuddies",
            },
            "ref": null,
            "rendered": "My Buddies",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "My Toolbox",
              "className": "toolbar-link",
              "replace": false,
              "to": "/toolbox",
            },
            "ref": null,
            "rendered": "My Toolbox",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "My Profile",
              "className": "toolbar-link",
              "replace": false,
              "to": "/profile",
            },
            "ref": null,
            "rendered": "My Profile",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Use the tabs below to connect with professionals or buddies.",
          "className": "dash-instruction",
        },
        "ref": null,
        "rendered": "Use the tabs below to connect with professionals or buddies.",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Tab
              label="Search Professionals"
>
              <SearchProfessionals
                            currentUser={
                                          Object {
                                                        "aboutMe": "Lola loves doggos.",
                                                        "city": "Denver",
                                                        "email": "lola@gmail.com",
                                                        "userName": "lolakoala",
                                                      }
                            }
                            getInsuranceList={[Function]}
                            getSpecialtyList={[Function]}
                            insuranceList={
                                          Array [
                                                        "medicaid",
                                                        "bcbs",
                                                        "aetna",
                                                      ]
                            }
                            profSearch={undefined}
                            pushHistory={[Function]}
                            search={[Function]}
                            specialtyList={
                                          Array [
                                                        "couples",
                                                        "youth",
                                                        "gender",
                                                      ]
                            }
              />
</Tab>,
            <Tab
              label="Search Buddies"
>
              <SearchBuddies
                            allChallenges={
                                          Array [
                                                        "anxiety",
                                                        "depression",
                                                      ]
                            }
                            buddySearch={undefined}
                            currentUser={
                                          Object {
                                                        "aboutMe": "Lola loves doggos.",
                                                        "city": "Denver",
                                                        "email": "lola@gmail.com",
                                                        "userName": "lolakoala",
                                                      }
                            }
                            getAllChallenges={[Function]}
                            pushHistory={[Function]}
                            search={[Function]}
              />
</Tab>,
          ],
          "initialSelectedIndex": 0,
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <SearchProfessionals
                currentUser={
                                Object {
                                                "aboutMe": "Lola loves doggos.",
                                                "city": "Denver",
                                                "email": "lola@gmail.com",
                                                "userName": "lolakoala",
                                              }
                }
                getInsuranceList={[Function]}
                getSpecialtyList={[Function]}
                insuranceList={
                                Array [
                                                "medicaid",
                                                "bcbs",
                                                "aetna",
                                              ]
                }
                profSearch={undefined}
                pushHistory={[Function]}
                search={[Function]}
                specialtyList={
                                Array [
                                                "couples",
                                                "youth",
                                                "gender",
                                              ]
                }
/>,
              "label": "Search Professionals",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "currentUser": Object {
                  "aboutMe": "Lola loves doggos.",
                  "city": "Denver",
                  "email": "lola@gmail.com",
                  "userName": "lolakoala",
                },
                "getInsuranceList": [Function],
                "getSpecialtyList": [Function],
                "insuranceList": Array [
                  "medicaid",
                  "bcbs",
                  "aetna",
                ],
                "profSearch": undefined,
                "pushHistory": [Function],
                "search": [Function],
                "specialtyList": Array [
                  "couples",
                  "youth",
                  "gender",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <SearchBuddies
                allChallenges={
                                Array [
                                                "anxiety",
                                                "depression",
                                              ]
                }
                buddySearch={undefined}
                currentUser={
                                Object {
                                                "aboutMe": "Lola loves doggos.",
                                                "city": "Denver",
                                                "email": "lola@gmail.com",
                                                "userName": "lolakoala",
                                              }
                }
                getAllChallenges={[Function]}
                pushHistory={[Function]}
                search={[Function]}
/>,
              "label": "Search Buddies",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allChallenges": Array [
                  "anxiety",
                  "depression",
                ],
                "buddySearch": undefined,
                "currentUser": Object {
                  "aboutMe": "Lola loves doggos.",
                  "city": "Denver",
                  "email": "lola@gmail.com",
                  "userName": "lolakoala",
                },
                "getAllChallenges": [Function],
                "pushHistory": [Function],
                "search": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Toolbar
            noGutter={false}
            style={
                        Object {
                                    "backgroundColor": "#d4c9a9",
                                  }
            }
>
            <Link
                        className="toolbar-link"
                        replace={false}
                        to="/chat"
            >
                        Chat
            </Link>
            <Link
                        className="toolbar-link"
                        replace={false}
                        to="/list/userProfs"
            >
                        My Preferred Professionals
            </Link>
            <Link
                        className="toolbar-link"
                        replace={false}
                        to="/list/userBuddies"
            >
                        My Buddies
            </Link>
            <Link
                        className="toolbar-link"
                        replace={false}
                        to="/toolbox"
            >
                        My Toolbox
            </Link>
            <Link
                        className="toolbar-link"
                        replace={false}
                        to="/profile"
            >
                        My Profile
            </Link>
</Toolbar>,
          <div
            className="dash-instruction"
>
            Use the tabs below to connect with professionals or buddies.
</div>,
          <Tabs
            initialSelectedIndex={0}
            onChange={[Function]}
>
            <Tab
                        label="Search Professionals"
            >
                        <SearchProfessionals
                                    currentUser={
                                                Object {
                                                            "aboutMe": "Lola loves doggos.",
                                                            "city": "Denver",
                                                            "email": "lola@gmail.com",
                                                            "userName": "lolakoala",
                                                          }
                                    }
                                    getInsuranceList={[Function]}
                                    getSpecialtyList={[Function]}
                                    insuranceList={
                                                Array [
                                                            "medicaid",
                                                            "bcbs",
                                                            "aetna",
                                                          ]
                                    }
                                    profSearch={undefined}
                                    pushHistory={[Function]}
                                    search={[Function]}
                                    specialtyList={
                                                Array [
                                                            "couples",
                                                            "youth",
                                                            "gender",
                                                          ]
                                    }
                        />
            </Tab>
            <Tab
                        label="Search Buddies"
            >
                        <SearchBuddies
                                    allChallenges={
                                                Array [
                                                            "anxiety",
                                                            "depression",
                                                          ]
                                    }
                                    buddySearch={undefined}
                                    currentUser={
                                                Object {
                                                            "aboutMe": "Lola loves doggos.",
                                                            "city": "Denver",
                                                            "email": "lola@gmail.com",
                                                            "userName": "lolakoala",
                                                          }
                                    }
                                    getAllChallenges={[Function]}
                                    pushHistory={[Function]}
                                    search={[Function]}
                        />
            </Tab>
</Tabs>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Link
                className="toolbar-link"
                replace={false}
                to="/chat"
>
                Chat
</Link>,
              <Link
                className="toolbar-link"
                replace={false}
                to="/list/userProfs"
>
                My Preferred Professionals
</Link>,
              <Link
                className="toolbar-link"
                replace={false}
                to="/list/userBuddies"
>
                My Buddies
</Link>,
              <Link
                className="toolbar-link"
                replace={false}
                to="/toolbox"
>
                My Toolbox
</Link>,
              <Link
                className="toolbar-link"
                replace={false}
                to="/profile"
>
                My Profile
</Link>,
            ],
            "noGutter": false,
            "style": Object {
              "backgroundColor": "#d4c9a9",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Chat",
                "className": "toolbar-link",
                "replace": false,
                "to": "/chat",
              },
              "ref": null,
              "rendered": "Chat",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "My Preferred Professionals",
                "className": "toolbar-link",
                "replace": false,
                "to": "/list/userProfs",
              },
              "ref": null,
              "rendered": "My Preferred Professionals",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "My Buddies",
                "className": "toolbar-link",
                "replace": false,
                "to": "/list/userBuddies",
              },
              "ref": null,
              "rendered": "My Buddies",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "My Toolbox",
                "className": "toolbar-link",
                "replace": false,
                "to": "/toolbox",
              },
              "ref": null,
              "rendered": "My Toolbox",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "My Profile",
                "className": "toolbar-link",
                "replace": false,
                "to": "/profile",
              },
              "ref": null,
              "rendered": "My Profile",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Use the tabs below to connect with professionals or buddies.",
            "className": "dash-instruction",
          },
          "ref": null,
          "rendered": "Use the tabs below to connect with professionals or buddies.",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Tab
                label="Search Professionals"
>
                <SearchProfessionals
                                currentUser={
                                                Object {
                                                                "aboutMe": "Lola loves doggos.",
                                                                "city": "Denver",
                                                                "email": "lola@gmail.com",
                                                                "userName": "lolakoala",
                                                              }
                                }
                                getInsuranceList={[Function]}
                                getSpecialtyList={[Function]}
                                insuranceList={
                                                Array [
                                                                "medicaid",
                                                                "bcbs",
                                                                "aetna",
                                                              ]
                                }
                                profSearch={undefined}
                                pushHistory={[Function]}
                                search={[Function]}
                                specialtyList={
                                                Array [
                                                                "couples",
                                                                "youth",
                                                                "gender",
                                                              ]
                                }
                />
</Tab>,
              <Tab
                label="Search Buddies"
>
                <SearchBuddies
                                allChallenges={
                                                Array [
                                                                "anxiety",
                                                                "depression",
                                                              ]
                                }
                                buddySearch={undefined}
                                currentUser={
                                                Object {
                                                                "aboutMe": "Lola loves doggos.",
                                                                "city": "Denver",
                                                                "email": "lola@gmail.com",
                                                                "userName": "lolakoala",
                                                              }
                                }
                                getAllChallenges={[Function]}
                                pushHistory={[Function]}
                                search={[Function]}
                />
</Tab>,
            ],
            "initialSelectedIndex": 0,
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <SearchProfessionals
                  currentUser={
                                    Object {
                                                      "aboutMe": "Lola loves doggos.",
                                                      "city": "Denver",
                                                      "email": "lola@gmail.com",
                                                      "userName": "lolakoala",
                                                    }
                  }
                  getInsuranceList={[Function]}
                  getSpecialtyList={[Function]}
                  insuranceList={
                                    Array [
                                                      "medicaid",
                                                      "bcbs",
                                                      "aetna",
                                                    ]
                  }
                  profSearch={undefined}
                  pushHistory={[Function]}
                  search={[Function]}
                  specialtyList={
                                    Array [
                                                      "couples",
                                                      "youth",
                                                      "gender",
                                                    ]
                  }
/>,
                "label": "Search Professionals",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "currentUser": Object {
                    "aboutMe": "Lola loves doggos.",
                    "city": "Denver",
                    "email": "lola@gmail.com",
                    "userName": "lolakoala",
                  },
                  "getInsuranceList": [Function],
                  "getSpecialtyList": [Function],
                  "insuranceList": Array [
                    "medicaid",
                    "bcbs",
                    "aetna",
                  ],
                  "profSearch": undefined,
                  "pushHistory": [Function],
                  "search": [Function],
                  "specialtyList": Array [
                    "couples",
                    "youth",
                    "gender",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <SearchBuddies
                  allChallenges={
                                    Array [
                                                      "anxiety",
                                                      "depression",
                                                    ]
                  }
                  buddySearch={undefined}
                  currentUser={
                                    Object {
                                                      "aboutMe": "Lola loves doggos.",
                                                      "city": "Denver",
                                                      "email": "lola@gmail.com",
                                                      "userName": "lolakoala",
                                                    }
                  }
                  getAllChallenges={[Function]}
                  pushHistory={[Function]}
                  search={[Function]}
/>,
                "label": "Search Buddies",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "allChallenges": Array [
                    "anxiety",
                    "depression",
                  ],
                  "buddySearch": undefined,
                  "currentUser": Object {
                    "aboutMe": "Lola loves doggos.",
                    "city": "Denver",
                    "email": "lola@gmail.com",
                    "userName": "lolakoala",
                  },
                  "getAllChallenges": [Function],
                  "pushHistory": [Function],
                  "search": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
